src/App.vue

```diff
@@ -40,12 +40,13 @@ export default {
         darkModePreference.addEventListener("change", () => {
             this.setTheme();
         });
-        if (this.getPreferenceBoolean("watchHistory", false))
-            if ("indexedDB" in window) {
-                const request = indexedDB.open("piped-db", 2);
-                request.onupgradeneeded = ev => {
-                    const db = request.result;
-                    console.log("Upgrading object store.");
+
+        if ("indexedDB" in window) {
+            const request = indexedDB.open("piped-db", 3);
+            request.onupgradeneeded = ev => {
+                const db = request.result;
+                console.log("Upgrading object store.");
+                if (this.getPreferenceBoolean("watchHistory", false)) {
```

Comment:
I feel we should create the store even if it not enabled, since this code is essentially ran once since its a migration.